lib/screens/new_question_screens/new_question_form.dart

```diff
@@ -174,23 +175,21 @@ class _NewQuestionFormState extends State<NewQuestionForm> {
     /// Create new Object to be sent to backend.
     setState(() {
       _question = new QuestionModel(
-        id: _questionId,
-        title: _questionTitle,
-        questionDate:
-            _date.toIso8601String(), // format date to add `T` character
-        description: _questionBody,
-        mediaUrls: downloadUrls,
-        category: widget.categoryName,
-        status: true,
-        visible: true,
-      );
+          id: _questionId,
+          title: _questionTitle,
+          questionDate:
+              _date.toIso8601String(), // format date to add `T` character
+          description: _questionBody,
+          mediaUrls: downloadUrls,
+          category: widget.categoryName,
+          status: true,
+          visible: true,
+          userId: EmailAuth().getCurrentUser().uid);
+      _question.toJson();
+      NewQuestionService()
+          .postNewQuestion(_question); // POST question to database
```

Comment:
These lines should be outside the setState() and it should have a try {} catch (e) {} block for it. Take a look at profile_onboarding.dart to see how it should be implemented. 

The service request should also be awaiting for it to complete: 
```suggestion
      _question.toJson();
      await NewQuestionService()
          .postNewQuestion(_question); // POST question to database
```

Reply:
I agree with Tony, you would need to put this new object you're creating in a try catch. Just in case the object does not get created.

Reply:
Done